codesign/export.go

```diff
@@ -1,21 +1,106 @@
 	"github.com/bitrise-io/go-utils/command"
 	"github.com/bitrise-io/go-utils/log"
-	"github.com/bitrise-tools/codesigndoc/osxkeychain"
-	"github.com/bitrise-tools/go-xcode/certificateutil"
-	"github.com/bitrise-tools/go-xcode/profileutil"
+	"github.com/bitrise-io/go-xcode/certificateutil"
+	"github.com/bitrise-io/go-xcode/profileutil"
+	"github.com/bitrise-io/goinp/goinp"
 )
 
-// CollectAndExportIdentities exports the given certificates into the given directory as a single .p12 file
-func CollectAndExportIdentities(certificates []certificateutil.CertificateInfoModel, absExportOutputDirPath string, isAskForPassword bool) error {
+// UploadConfig contains configuration to automatically upload artifacts to bitrise.io
+type UploadConfig struct {
+	PersonalAccessToken string
+	AppSlug             string
+}
+
+// UploadAndWriteCodesignFiles exports then uploads codesign files to bitrise.io and saves them to output folder
+func UploadAndWriteCodesignFiles(certificates []certificateutil.CertificateInfoModel, profiles []profileutil.ProvisioningProfileInfoModel, askForPassword bool, outputDirPath string, uploadConfig UploadConfig) (bool, bool, error) {
+	identities, err := collectAndExportIdentities(certificates, askForPassword)
+	if err != nil {
+		return false, false, err
+	}
+
+	provisioningProfiles, err := collectAndExportProvisioningProfiles(profiles)
+	if err != nil {
+		return false, false, err
+	}
+
+	var client *bitrise.Client
+	if uploadConfig.PersonalAccessToken != "" && uploadConfig.AppSlug == "" ||
+		uploadConfig.PersonalAccessToken == "" && uploadConfig.AppSlug != "" {
+		log.Warnf(`Both auth-token and app-slug need to be set for automatic upload.
+		Proceeding with interactive prompts for personal access token and app slug.`)
+	}
+	if uploadConfig.PersonalAccessToken != "" && uploadConfig.AppSlug != "" {
+		// Upload automatically if token is provided as CLI paramter, do not export to filesystem
+		// Used to upload artifacts as part of an other CLI tool
+		client, err = bitrise.NewClient(uploadConfig.PersonalAccessToken)
+		if err != nil {
+			return false, false, err
+		}
+		client.SetSelectedAppSlug(uploadConfig.AppSlug)
+	}
+
+	if client == nil {
+		uploadConfirmMsg := "Do you want to upload the provisioning profiles and certificates to Bitrise?"
+		if len(provisioningProfiles) == 0 {
+			uploadConfirmMsg = "Do you want to upload the certificates to Bitrise?"
+		}
+		fmt.Println()
+		if shouldUpload, err := goinp.AskForBoolFromReader(uploadConfirmMsg, os.Stdin); err != nil {
+			return false, false, err
+		} else if shouldUpload {
+			client, err = bitriseio.GetInteractiveConfigClient()
+		}
+	}
+
+	provProfilesUploaded := (len(profiles) == 0)
+	certsUploaded := (len(certificates) == 0)
+	if client != nil {
+		certsUploaded, provProfilesUploaded, err = bitriseio.UploadCodesigningFiles(client, identities, provisioningProfiles)
+		if err != nil {
+			return false, false, err
+		}
+	}
+
+	if strings.TrimSpace(outputDirPath) != "" {
```

Comment:
is `outputDirPath` always expanded? current dir can be passed by an empty string, would not it case issue?

Reply:
It is at the moment, in initExportOutputDir()